substrate-archive-common/src/database.rs

```diff
@@ -0,0 +1,13 @@
+use super::error::Result;
+
+pub type KeyValuePair = (Box<[u8]>, Box<[u8]>);
+
+// Archive specific K/V database reader implementation
+pub trait ReadOnlyDatabaseTrait: Send + Sync {
```

Comment:
There is a minor naming inconsistency with imported substrate code here which uses `NameT` rather than `NameTrait` for traits.